frictionless/portals/github/manager.py

```diff
@@ -1,45 +1,156 @@
-            else:
-                paths.append(file_content.path)
-        package = Package()
-        for path in paths:
-            fullpath = f"https://raw.githubusercontent.com/{location}/{branch}/{path}"
-            if path in ["datapackage.json", "datapackage.yaml"]:
-                return Package.from_descriptor(fullpath)
-            if path.endswith(".csv"):
-                package.add_resource(Resource(path=fullpath))
-        return package
+        paths = _get_contents(contents, repository)
+        package = _get_package(paths, repository, base_path, self.control.formats)
+        if package.resources:
+            return package
+        note = "Package not found"
+        raise FrictionlessException(note)
 
     # Write
-    # TODO: implement
+
+    def write_package(
+        self,
+        *,
+        email: Optional[str] = None,
+        repo: Optional[str] = None,
+        apikey: Optional[str] = None,
+        filename: Optional[str] = None,
+        content: Optional[str] = None,
+        branch: Optional[str] = None,
+    ):
+        repo = repo or self.control.repo
+        apikey = apikey or os.environ.get("GITHUB_ACCESS_TOKEN", None)
+        assert repo
+        assert apikey
+        # Create repo
+        repository = None
+        user = None
+        try:
+            client = platform.github.Github(apikey)
+            user = client.get_user()
+            repository = user.create_repo(
+                name=repo, auto_init=True, gitignore_template="Python"
+            )
+        except Exception as exception:
+            note = "Github API error:" + repr(exception)
+            raise FrictionlessException(note)
+
+        # Write package file to the repo
+        file_created = None
+        if content:
+            newfile = filename or "package.json"
+            repository = user.get_repo(repo)
+            email = (
+                user.email or self.control.email or os.environ.get("GITHUB_EMAIL", None)
+            )
+            username = (
+                user.name or self.control.user or os.environ.get("GITHUB_NAME", None)
+            )
+            assert email
+            assert username
+            try:
+                author = platform.github.InputGitAuthor(username, email)
+                branch = branch or repository.default_branch
+                file_created = repository.create_file(
+                    path=newfile,
+                    message="Create package.json",
+                    content=content,
+                    branch=branch,
+                    committer=author,
+                    author=author,
+                )
+            except Exception as exception:
+                note = "Github API error:" + repr(exception)
+                raise FrictionlessException(note)
+
+        response = namedtuple("response", ["repository", "file_created"])
+        return response(repository, file_created)
+
+
+def _get_contents(contents, repository):
```

Comment:
I would make this function `get_resources() -> List[Resource]` as on the next step we also need to add some metadata like `resource.name`. Probably the best way for doing so will be:

```
resource = Resource(path=path)
resource.infer(sample=False) # it means infer anything you can without any file reading
```

Reply:
But if Github API provides metdata like scheme/format/mediatype etc we need to set it directly

```
resource = Resource(path=path, format=format, ...)
```

Reply:
I used resource.infer as I couldn't find any function that would return these file metadata.

get_content is returning all the paths that also includes datapackage.json or datapackage.yaml so was not sure if it could be included as resource get_resources() -> List[Resource]. So I changed the name of the function but return type set as list[str]